tests/unit/acl-v2.tcl

```diff
@@ -0,0 +1,58 @@
+start_server {tags {"acl external:skip"}} {
+    set r2 [redis_client]
+    test {Test basic multiple selectors} {
+        r ACL SETUSER selector-1 on -@all resetkeys nopass
+        $r2 auth selector-1 password
+        catch {$r2 ping} _
+        catch {$r2 set write::foo bar} _
+        catch {$r2 get read::foo} _
+
+        r ACL SETUSER selector-1 (+@write ~write::*) (+@read ~read::*)
+        catch {$r2 ping} _
+        assert_equal "OK" [$r2 set write::foo bar]
+        assert_equal "" [$r2 get read::foo]
+        catch {$r2 get write::foo} _
+        catch {$r2 set read::foo bar} _
+    }
+
+    test {Test flexible selector definition} {
+        r ACL SETUSER selector-2 (~key1 +get ) ( ~key2 +get ) ( ~key3 +get) (~key4 +get)
+        set get_user_result [r ACL GETUSER selector-2]
+        assert_equal "" [lindex [lindex [lindex [lindex $get_user_result 11] 0] 3] 0]
+        assert_equal "key1" [lindex [lindex [lindex [lindex $get_user_result 11] 1] 3] 0]
+        assert_equal "key2" [lindex [lindex [lindex [lindex $get_user_result 11] 2] 3] 0]
+        assert_equal "key3" [lindex [lindex [lindex [lindex $get_user_result 11] 3] 3] 0]
+        assert_equal "key4" [lindex [lindex [lindex [lindex $get_user_result 11] 4] 3] 0]
```

Comment:
would be better to convert the response to dict, and fetch the intended fields, rather than rely on hard coded indexes.